astrokat/utility.py

```diff
@@ -133,7 +137,9 @@ def get_lst(yaml_lst):
         start_lst = time_.hour + time_.minute/60.
 
     if end_lst is None:
-        end_lst = (start_lst + 12.) % 24.
+        end_lst = (start_lst + 24.) % 24.
+        if numpy.abs(end_lst - start_lst) < 1.:
+            end_lst = 24.
```

Comment:
Why is this done?  If start and end LST are within 1 hour, then force end LST to midnight?
Looks like it will often be the case, e.g. start LST values from `[0..24)` would end up here.
Not obvious for a non-astronomer like me 🙂

Reply:
It is to stabilise the math -- mod of 0 (zero) give zero, then end and start is both zero and nothing happens, even though you want it to run over 24 hours

Plus, I need to include 24 -- even though technically I never reach it -- I had some commissioning astronomers complain if I do not set this to explicit 24

Reply:
So if unspecified, end LST is always 24, unless start LST >= 24:
```python
In [1]: import numpy

In [2]: def calc_end_lst(start_lst):
   ...:     end_lst = (start_lst + 24.) % 24.
   ...:     if numpy.abs(end_lst - start_lst) < 1.:
   ...:         end_lst = 24.
   ...:     return end_lst
   ...: 

In [3]: for start_lst in [0., 1., 6., 12., 18., 24., 30.]:
   ...:     end_lst = calc_end_lst(start_lst)
   ...:     print('start_lst {} .. end_lst {}'.format(start_lst, end_lst))
   ...:     
start_lst 0.0 .. end_lst 24.0
start_lst 1.0 .. end_lst 24.0
start_lst 6.0 .. end_lst 24.0
start_lst 12.0 .. end_lst 24.0
start_lst 18.0 .. end_lst 24.0
start_lst 24.0 .. end_lst 0.0
start_lst 30.0 .. end_lst 6.0
```

Reply:
Yes, exactly, LST = 0 - 24 floating hours (decimal).
Normally a single target LST range is 8 to 10 hours, visible above the horizon, which is 4 minutes earlier each day (celestial time), thus %24 

But zero (0) is never our friend, so special checks need to be implemented if you have a mod value that is a factor of 24, but close to the a full 24 hour time difference.

When will this happen?
Normally if we have targets that has been chosen to always have at least one target visible no matter what time in a 24 hour cycle the observation get scheduled.

So in short, one of those nasty times when astronomy may cause our math to fail in practise, but still be valid in theory....